hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -112,6 +116,15 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     }
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+    TableOperations ops = newTableOps(conf(), identifier);
+    HadoopInputFile metadataFile = HadoopInputFile.fromLocation(metadataFileLocation, conf());
+    TableMetadata metadata = TableMetadataParser.read(ops, metadataFile);
+    ops.register(metadata);
```

Comment:
What about using `ops.commit(null, metadata)` instead? Then `commit` could detect that the metadata is already written because `metadata.file()` is non-null and use that path instead of writing the metadata file. I think that would be cleaner than adding a new method to `TableOperations`.